discovery-provider/src/utils/session_manager.py

```diff
@@ -11,10 +14,24 @@
 class SessionManager:
     def __init__(self, db_url, db_engine_args):
         self._engine = create_engine(db_url, **db_engine_args)
+
+        @event.listens_for(Engine, "before_cursor_execute", retval=True)
+        def comment_sql_calls(
+            conn, cursor, statement, parameters, context, executemany
+        ):
+            if "src" in conn.info:
+                statement = statement + " -- %s" % conn.info.pop("src")
+            return statement, parameters
+
         self._session_factory = sessionmaker(bind=self._engine)
         # Attach a listener for new engine connection.
         # See https://docs.sqlalchemy.org/en/14/core/event.html
         listen(self._engine, "connect", self.on_connect)
+        listen(self._session_factory, "after_begin", self.session_on_after_begin)
+
+    def session_on_after_begin(self, session, transaction, connection):
```

Comment:
What's this doing? I'm unclear on this functionality

Reply:
New comments should help explain.